samcli/commands/local/cli_common/options.py

```diff
@@ -48,7 +48,24 @@ def local_common_options(f):
             default=False,
             help="If set, will emulate a shutdown event after the invoke completes, "
             "in order to test extension handling of shutdown behavior.",
-        )
+        ),
+        click.option(
+            "--container-host",
+            default="localhost",
+            show_default=True,
+            help="Host of locally emulated Lambda container. "
+            "This option is useful when the container runs on a different host than SAM CLI. "
+            "For example, if you want to run SAM CLI in a Docker container on macOS, "
+            "use this option with host.docker.internal",
+        ),
+        click.option(
+            "--container-host-interface",
+            default="127.0.0.1",
+            show_default=True,
+            help="Interface that Docker host binds ports to. "
+            "This option is useful when users want to bind ports to a different interface. "
+            "Note: the input should be an IP address. ",
```

Comment:
Also mention that `0.0.0.0` binds to all interfaces?

Reply:
Sure!